plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,718 @@
+            target=target,
+            size=size,
+            space_type=space_type,
+            cmd="",
+            changed=changed,
+            rc=1,
+            stdout="",
+            stderr="",
+            verbose_output="",
+        )
+    )
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    # Validate if the target ZFS exist
+    if not (data_set.DataSet.data_set_exists(target)):
+        module.fail_json(msg="ZFS Target {0} does not exist".format(target), **result)
+
+    # Validation to found target on the system and also get the mount_point
+    mount_target = find_mount_target(module=module, target=target, results=result)
+
+    if size <= 0:
+        module.fail_json(msg="Can not resize ZFS Target {0} to 0".format(target), **result)
+
+    # Initialize the class with the target
+    zfsadm_obj = zfsadm(aggregate_name=target, module=module)
+
+    rc, stdout, stderr = zfsadm.get_aggregate_size(zfsadm_obj.aggregate_name, module)
+
+    if rc == 0:
+        old_size, old_free = get_size_and_free(line=stdout)
+
+    if space_type != "k":
+        space = calculate_size_on_k(size=size, space_type=space_type)
+    else:
+        space = size
+
+    # Validations to know witch function will be execute
+    operation = ""
+    minimum_size_t_shrink = old_size - old_free
+
+    if space_type != "k":
+        size_on_type = convert_size(size=old_size, space_type=space_type)
+        free_on_type = convert_size(size=old_free, space_type=space_type)
+
+    str_old_size = old_size if space_type == "k" else size_on_type
+    str_old_size = proper_size_str(str_old_size, space_type)
+    str_old_free = old_free if space_type == "k" else free_on_type
+    str_old_free = proper_size_str(str_old_free, space_type)
+
+    result.update(
+        dict(
+            mount_target=mount_target,
+            old_size=str_old_size,
+            old_free_space=str_old_free,
+        )
+    )
+
+    if space == old_size:
+        result.update(
+            dict(
+                cmd="",
+                rc=0,
+                stdout="Size provided is the current size of the ZFS {0}".format(target),
+                stderr="",
+                changed=False,
+                size=size,
+                new_size=str_old_size,
+                new_free_space=str_old_free,
+            )
+        )
+        module.exit_json(**result)
+
+    elif space < minimum_size_t_shrink:
+        module.fail_json(msg="Not enough free space in the ZFS to shrink.", **result)
+
+    elif space > old_size:
+        operation = "grow"
+
+    elif space >= minimum_size_t_shrink and space < old_size:
+        operation = "shrink"
+
+    noai = " -noai " if noai else ""
+    noai = "" if operation == "grow" else noai
+
+    # Variables for the verbose output or trace destination
+    trace = ""
+    tmp_file = ""
+    trace_uss = True
+    trace_destination_created = True
+
+    if trace_destination is not None:
+        if "/" in trace_destination:
```

Comment:
While I realize the docs state absolute path and not relative, if a user entered `~/u/xyz` I think on line 590:
```
trace = f" -trace '{tmp_file}'" if trace_uss else f" -trace \"//'{trace_destination}'\" "
```
you will have a problem because the utility expects absolute paths and while the check for `/` is not enough, as it does not try to expand `~/` , have a look at the `better_arg_parser.py` , [here](https://github.com/ansible-collections/ibm_zos_core/blob/23efb55fae9b39012519512beb55e6de57ff255b/plugins/module_utils/better_arg_parser.py#L668), you should be able to use that method, see how our other code is using it.

Reply:
What about a relative path as in `.somefile` , it would get missed by this check `if "/" in trace_destination:`, the better arg parser will fix it but you have to send it to the better arg parser , I think `if "/" in trace_destination or "." in trace_destination` would take care of that.